File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/implement/PostCommentUpdater.java
package com.eager.questioncloud.core.domain.post.implement;

import com.eager.questioncloud.core.domain.post.model.PostComment;
import com.eager.questioncloud.core.domain.post.repository.PostCommentRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class PostCommentUpdater {
    private final PostCommentRepository postCommentRepository;

    public void modify(Long commentId, Long userId, String comment) {
        PostComment postComment = postCommentRepository.findByIdAndWriterId(commentId, userId);
        postComment.modify(comment);
        postCommentRepository.save(postComment);
    }
}

File: qc-core/qc-post/qc-post-api/src/main/kotlin/com/eager/questioncloud/post/implement/PostCommentUpdater.kt
package com.eager.questioncloud.post.implement

import com.eager.questioncloud.post.command.ModifyPostCommentCommand
import com.eager.questioncloud.post.repository.PostCommentRepository
import org.springframework.stereotype.Component

@Component
class PostCommentUpdater(
    private val postCommentRepository: PostCommentRepository,
) {
    fun modify(command: ModifyPostCommentCommand) {
        val postComment = postCommentRepository.findByIdAndWriterId(command.commentId, command.userId)
        postComment.modify(command.comment)
        postCommentRepository.save(postComment)
    }
}